types/src/crypto/asymmetric_key.rs

```diff
@@ -283,13 +287,13 @@ impl FromBytes for SecretKey {
             ED25519_TAG => {
                 let (raw_bytes, remainder): ([u8; Self::ED25519_LENGTH], _) =
                     FromBytes::from_bytes(remainder)?;
-                let secret_key = Self::ed25519(raw_bytes);
+                let secret_key = Self::ed25519(raw_bytes).unwrap();
```

Comment:
This method already returns a `Result` - why are we unwrapping here and just below?